semeio/jobs/ots/ots.py

```diff
@@ -0,0 +1,477 @@
+                        r1 = (surface.x[point], surface.y[point], rz)
+                        r2 = (surface.x[point], surface.y[point], rz + 0.1)
+                        # subsidence and displacement have opposite sign
+                        # should have minus here, more efficient
+                        # when calculating ts
+                        dz1 = self.subsidence.evalGeertsma(
+                            base_survey="base",
+                            monitor_survey="monitor",
+                            pos=r1,
+                            youngs_modulus=self._youngs_modulus,
+                            poisson_ratio=self._poisson_ratio,
+                            seabed=self._seabed,
+                        )
+                        dz2 = self.subsidence.evalGeertsma(
+                            base_survey="base",
+                            monitor_survey="monitor",
+                            pos=r2,
+                            youngs_modulus=self._youngs_modulus,
+                            poisson_ratio=self._poisson_ratio,
+                            seabed=self._seabed,
+                        )
+
+                        ezz = (dz2 - dz1) * 10
+                        surf_displacement[point] += ezz
+
+            self._report("TS", pair[0], pair[1], len(points_to_calculate))
+
+            # subsidence and displacement have opposite sign, thus the minus sign
+            ts = -self._r_factor * surf_displacement * surface.dt * 1000
+            self._divide_negative_shift(ts)
+
+            ts = ts * self._convention
+
+            ts_surfaces.append(self._create_surface(ts))
+
+        return ts_surfaces
+
+    @staticmethod
+    def _vintages_name_date(vintage_pairs):
+        vintages = set()
+        for pair in vintage_pairs:
+            vintages.add(pair[0])
+            vintages.add(pair[1])
+        vintages_date = list(vintages)
+        vintages_name = []
+        for i in range(len(vintages_date)):
+            vintages_name.append("S{}".format(i))
+
+        Vintages = namedtuple("Vintages", "name date")
+        return Vintages(vintages_name, vintages_date)
+
+    def _report(self, func_name, base, monitor, num_points_calculated):
+        if self._convention == 1:
+            start_date, end_date = monitor, base
+        elif self._convention == -1:
+            start_date, end_date = base, monitor
+        else:
+            raise ValueError(
+                "Convention must be 1 or -1, was {}".format(self._convention)
+            )
+        logging.debug(
+            "{:%x %X} {}: Calculating shift"
+            " {:%Y.%m.%d}-{:%Y.%m.%d} in {} points".format(
+                dt.now(), func_name, start_date, end_date, num_points_calculated
+            )
+        )
+
+    def _get_non_nan_points(self):
+        points_to_calculate = [
+            id for id in range(len(self._surface)) if not np.isnan(self._surface.z[id])
```

Comment:
Overriding built-in function https://docs.python.org/3/library/functions.html#id

Although it will be locally it is bad behaviour..

Reply:
Good point! Had no idea that `id` is taken.

Reply:
(I got a bit help from my IDE to be honest :P )